gstd/gstd.c

```diff
@@ -200,12 +125,14 @@ main (gint argc, gchar * argv[])
   g_option_context_add_main_entries (context, entries, NULL);
 
   /* Initialize GStreamer */
-  gstreamer_group = gst_init_get_option_group ();
+  gstd_manager_new (supported_ipcs, num_ipcs, &manager,
+      &gstreamer_group, 0, NULL);
   g_option_context_add_group (context, gstreamer_group);
 
   /* Read option group for each IPC */
-  ipc_add_option_groups (ipc_array, supported_ipcs, num_ipcs, context,
-      optiongroup_array);
+  ipc_group_array = g_malloc0 (num_ipcs * sizeof (*ipc_group_array));
+  gstd_manager_ipc_options (manager, ipc_group_array);  /* If you don't want this option, you can avoid calling this function */
+  g_option_context_add_group (context, *ipc_group_array);
```

Comment:
If `ipc_group_array` is an array shouldn't we call `g_option_context_add_group` for each element in the array?